test/micro/org/openjdk/bench/vm/runtime/NMTBenchmark_wb.java

```diff
@@ -0,0 +1,147 @@
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package org.openjdk.bench.vm.runtime;
+
+import org.openjdk.bench.vm.lang.Throw;
+import org.openjdk.jmh.annotations.*;
+import org.openjdk.jmh.infra.BenchmarkParams;
+import org.openjdk.jmh.infra.Blackhole;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.concurrent.TimeUnit;
+import java.lang.*;
+import java.util.*;
+
+import jdk.internal.misc.Unsafe;
+import jdk.test.whitebox.WhiteBox;
+import org.openjdk.jmh.runner.Runner;
+import org.openjdk.jmh.runner.RunnerException;
+import org.openjdk.jmh.runner.options.Options;
+import org.openjdk.jmh.runner.options.OptionsBuilder;
+
+@BenchmarkMode(Mode.AverageTime)
+@OutputTimeUnit(TimeUnit.MILLISECONDS)
+@State(Scope.Benchmark)
+@Warmup(iterations = 1, time = 1, timeUnit = TimeUnit.SECONDS)
+@Measurement(iterations = 5, time = 1, timeUnit = TimeUnit.SECONDS)
+public abstract class NMTBenchmark_wb {
+
+    @Param({"2", "4", "8", "16"})
+    public int THREADS;
+
+    @Param({"16", "32", "64"})
+    public int REGIONS;
+
+    @Param({"4", "8", "16", "32", "64"})
+    public int SUB_REGIONS;
+
+    private static final int PageSize = 1024 * 4;
+
+    // Need to wrap WhiteBox in a holder class so that it doesn't get initialized on the host VM (which won't
+    // have WB enabled.
+    private static class WhiteBoxHolder {
+
+        private static final WhiteBox WB;
+
+        static {
+            System.out.println("Current user dir: " + System.getProperty("user.dir"));
+            WhiteBox wb = null;
+            try {
+                wb = WhiteBox.getWhiteBox();
+            } catch (Throwable t) {
+                System.out.println("WhiteBox initialization failure: " + t);
+                t.printStackTrace();
+                // Explicit exit to avoid initialization loops.
+                System.exit(17);
+            }
+            WB = wb;
+        }
+    }
```

Comment:
I got help from @cl4es for this part, when I have had problems using WB in JMH. 
It cannot be part of JMH, since not all benchmarks use WB. It should only be used for benchmarks that needs and use WB.
Ping @cl4es.